### PR TITLE
Add Create ResourceSet Snapshot endpoint

### PR DESCRIPTION
**Pull request by abhishekdharmapurikar**
# Description

Please describe your pull request in detail.

## Related Issue

This project only accepts pull requests related to open issues.

* If suggesting a new feature or change, please discuss it in an issue first.
* If fixing a bug, there should be an issue describing it with steps to reproduce

_Please link to the issue here_

## Motivation and Context

Why is this change required? What problem does it solve?

## How Has This Been Tested?

* Please describe in detail how you tested your changes.
* Include details of your testing environment, and the tests you ran to see how your change affects other areas of the code, etc.

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

Go over all the following points, and put an `x` in all the boxes that apply. If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the CHANGELOG file accordingly for the version that this merge modifies.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: pkg/polaris/graphql/infinity-k8s/infinity_k8s.go [hunk 1]
```diff
@@ -83,6 +83,10 @@ type ExportK8sResourceSetSnapshotJobConfig struct {
 	Filter              string `json:"filter,omitempty"`
 }
 
+type BaseOnDemandSnapshotConfigInput struct {
```
Comment (sai-sandeep-1) [inline pkg/polaris/graphql/infinity-k8s/infinity_k8s.go:86]: any reason behind this specific name? 
can't we just call this `OnDemandSnapshotJobConfig`?

File: pkg/polaris/graphql/infinity-k8s/infinity_k8s.go [hunk 2]
```diff
@@ -331,6 +335,45 @@ func (a API) ExportK8sResourceSetSnapshot(
 	return payload.Data.Response, nil
 }
 
+// CreateK8sResourceSnapshot takes a ResourceSetID, the snapshot job config and
```
Comment (sai-sandeep-1) [inline pkg/polaris/graphql/infinity-k8s/infinity_k8s.go]: `ResourceSetID` -> `ResourceSetFID`

**Review by sai-sandeep-1 — APPROVED** [review]

File: pkg/polaris/graphql/infinity-k8s/infinity_k8s.go (hunk 1 above)
Comment (abhishekdharmapurikar) [reply pkg/polaris/graphql/infinity-k8s/infinity_k8s.go:86]: Borrowed it from the graphql schema name. It is shared among various components. Should I rename to your suggestion?

File: pkg/polaris/graphql/infinity-k8s/infinity_k8s.go (hunk 2 above)
Comment (abhishekdharmapurikar) [reply pkg/polaris/graphql/infinity-k8s/infinity_k8s.go]: Ahh yes!

File: pkg/polaris/graphql/infinity-k8s/infinity_k8s.go (hunk 1 above)
Comment (sai-sandeep-1) [reply pkg/polaris/graphql/infinity-k8s/infinity_k8s.go:86]: Let's leave it to be the same as the graphql object name.

**Pull request merged**